_posts/2022-03-24-opensearch-java-runtime.markdown

```diff
@@ -0,0 +1,78 @@
+|:-----------------------------------------------------------------------------------------------------------------------------:|:------------------------------:|:---------------:|
+| [1.0.0](https://opensearch.org/versions/opensearch-1-0-0.html)-[1.2.4](https://opensearch.org/versions/opensearch-1-2-4.html) | AdoptOpenJDK 15.0.1+9          | 11, 14, 15      |
+| [1.3.0](https://opensearch.org/versions/opensearch-1-3-0.html)                                                                | Adoptium (Temurin) 11.0.14.1+1 | 8, 11, 14       |
+
+This, however, doesn't tell the whole story. The OpenSearch distribution is comprised of the engine and a dozen plugins. Both the engine, and each plugin, are built and tested with a range of JDKs (a subset of 8, 11 LTS, 14, 15, and 17), across multiple operating systems (Linux, Windows, FreeBSD, and MacOS). Then, the complete distribution is rebuilt from source, and tested with the bundled JDK. Finally, while we were claiming compatibility with JDK 8, we weren't running tests with that version for most plugins, nor were they actually built to target JDK8 in most components.
+
+### Versions 1.0 to 1.2.0
+
+The complete distribution of OpenSearch 1.0 through 1.2.4 was built with JDK 14, and tested with the bundled JDK 15. Various individual components were built and tested individually with JDK 14.
+
+### Version 1.3.0
+
+In 1.3.0 we have replaced version 14 with an LTS version 11 for both builds and releases. All components build and test with JDK 8, 11, and 14.
+
+The parent issues for this change were [opensearch-build#64](https://github.com/opensearch-project/opensearch-plugins/issues/64) and [opensearch-build#74](https://github.com/opensearch-project/opensearch-build/issues/74). The implementation in OpenSearch engine was [OpenSearch#940](https://github.com/opensearch-project/OpenSearch/pull/940), and was followed by plugins, e.g. [security#1580](https://github.com/opensearch-project/security/pull/1580). We also lowered the source and target Java versions back to 8 in [OpenSearch#2321](https://github.com/opensearch-project/OpenSearch/pull/2321) and fixed any incompatible code in the engine and plugins. Version 1.3.0 now reliably runs on JDK8. 
+
+Originally, we were also planning to upgrade the bundled JDK to 17 in this version, but ran into a number of issues. We decided to downgrade to JDK 11 in [OpenSearch#2301](https://github.com/opensearch-project/OpenSearch/pull/2301), and deferred upgrading to JDK 17 to 2.0.0 via [opensearch-plugins#129](https://github.com/opensearch-project/opensearch-plugins/issues/129).
+
+#### Customizing the OpenSearch Runtime
+
+By default, any OpenSearch distribution consults `JAVA_HOME` first in order to find out the JVM runtime to run on. If `JAVA_HOME` is not set, OpenSearch will try to fallback to the bundled JVM runtime if available. 
```

Comment:
I might be showing my ignorance, but I think you're conflating the JVM and JDK in this context. Doesn't `JAVA_HOME` point to a JDK not the JVM?

Reply:
Java runtime